src/com/senlainc/warsaw/tyurin/Test.java

```diff
@@ -0,0 +1,24 @@
+package com.senlainc.warsaw.tyurin;
+
+public class Test {
+
+    public static void main(String[] args) {
+
+        BodyLineStep bodyLineStep = new BodyLineStep();
+        Body body = (Body) bodyLineStep.buildProductPart();
+        System.out.println("Created part of tank - body");
+
+        EngineLineStep engineLineStep = new EngineLineStep();
+        Engine engine = (Engine) engineLineStep.buildProductPart();
```

Comment:
Same here, your StepLines produces abstract IProductPart, if you change variable type here to it class cast will not be needed